azure-iot-device/azure/iot/device/iothub/pipeline/pipeline_stages_iothub_mqtt.py

```diff
@@ -60,14 +61,17 @@ def _run_op(self, op):
             # The customer user agent string would simply be appended to the end of this username, in URL Encoded format.
             query_param_seq = [
                 ("api-version", pkg_constant.IOTHUB_API_VERSION),
-                ("DeviceClientType", ProductInfo.get_iothub_user_agent()),
+                (
+                    "DeviceClientType",
+                    ProductInfo.get_iothub_user_agent()
+                    + str(self.pipeline_root.pipeline_configuration.product_info),
```

Comment:
>+ str(self.pipeline_root.pipeline_configuration.product_info), [](start = 20, length = 62)

Is there any behavior change when you change the code to append product_info _before_ URL encoding?  Your code looks correct, I'm just wondering if you changed this to make it more correct or did you also intend a change in behavior?

Reply:
There is no change. This is just cleaner and clearer to the relationship between user agent and product info.

---
In reply to: [399580837](https://github.com/Azure/azure-iot-sdk-python/pull/514#discussion_r399580837) [](ancestors = 399580837)